### PR TITLE
Options for random goal boundaries

### PR DESCRIPTION
**Pull request by loopfz**
This PR introduces support to random instrument count ranges such as `--goal 5-8`

This lets you keep the mystery around the exact instrument count while having some control over the total seed length (e.g. avoiding 1-instrument seeds when you wanted something longer, or vice versa)

### CODE REVIEW

File: randomizer.py [hunk 1]
```diff
@@ -27,8 +27,8 @@ def __init__(self, options, *, seed=None):
         self.rnd = random.Random(self.seed)
         if options.race:
             self.rnd.random()  # Just pull 1 random number so race seeds are different then from normal seeds but still stable.
-        if options.goal == "random":
-            options.goal = self.rnd.randint(-1, 8)
+        if type(options.goal) is range:
```
Comment (daid) [inline randomizer.py:30]: Better use `isinstance()` instead of `type()`

File: www/options.php [hunk 2]
```diff
@@ -47,7 +47,7 @@
...
+        'goal' => ['label' => 'Goal', 'type' => ['8' => '8 instruments', '7' => '7 instruments', '6' => '6 instruments', '5' => '5 instruments', '4' => '4 instruments', '3' => '3 instruments', '2' => '2 instruments', '1' => '1 instrument', '0' => 'No instruments', '-1' => 'Egg already open', 'random' => 'Random instrument count', 'open-4' => 'Random short game (0-4)', '5-8' => 'Random long game (5-8)', 'seashells' => 'Seashell hunt (20)'], 'default' => '8', 'arg' => '--goal',
             'tooltip' => 'Changes the goal of the game.|[1-8 instruments], number of instruments required to open the egg.|[No instruments] open the egg without instruments, still requires the ocarina with the balled of the windfish|[Egg already open] the egg is already open, just head for it once you have the items needed to defeat the boss.|[Seashell hunt] egg will open once you collected 20 seashells. Instruments are replaced by seashells and shuffled.'],
```
Comment (daid) [inline www/options.php:51]: Short/long random are not explained in the tooltip.

File: randomizer.py (hunk 1 above)
Comment (loopfz) [reply randomizer.py:30]: fixing

File: www/options.php (hunk 2 above)
Comment (loopfz) [reply www/options.php:51]: Ah yes; since `random` wasnt explained either I assumed that was intentional to avoid clutter

**Pull request merged**